src/actors/objects/mod.rs

```diff
@@ -473,18 +461,24 @@ impl ObjectsActor {
                 .iter()
                 .map(|source| {
                     let type_name = source.type_name();
-
-                    prepare_downloads(&source, filetypes, &identifier)
-                        .or_else(move |e| {
-                            // This basically only happens for the Sentry source type, when doing the
-                            // search by debug/code id. We do not surface those errors to the user
-                            // (instead we default to an empty search result) and only report them
-                            // internally.
-                            log::error!("Failed to download from {}: {}", type_name, LogError(&e));
-                            Ok(Vec::new())
-                        })
-                        // create a new hub for each prepare_downloads call
-                        .sentry_hub_new_from_current()
+                    let hub = Hub::new_from_top(Hub::current());
+                    Hub::run(Arc::new(hub), || {
```

Comment:
Not that I want to change anything here now.  But this still makes me think it might be better to integrate the hub binding into the services API because it is rather deceptive that you can't just bind a hub to the future.  But perhaps there's something even better we can do with asyncfn.

Reply:
Agreed that we can just wait for `async fn`. That said, maybe add a comment to clarify why we choose this approach over `.bind_hub`.